Source/WebCore/platform/graphics/ColorSerialization.cpp

```diff
@@ -553,7 +553,10 @@ String serializationForCSS(const SRGBA<float>& color, bool useColorFunctionSeria
 
 String serializationForHTML(const SRGBA<float>& color, bool useColorFunctionSerialization)
 {
-    return serializationForCSS(color, useColorFunctionSerialization);
+    if (useColorFunctionSerialization)
+        return serializationUsingColorFunction(color);
+
+    return serializationForHTML(convertColor<SRGBA<uint8_t>>(color), false);
```

Comment:
Is there any issue with loss of precision here if values are not integers?

Reply:
No, this is what the spec requires. When precision is actually wanted, the new syntaxes (e.g. color(srgb ...)) maintain full precision. This is just weird legacy edge cases.